source/xml/valueobjects.md

```diff
@@ -0,0 +1,217 @@
+
+
+Writing
+-------
+
+Writing is similarly easy. Given that you have a `$feed` variable which refers
+to a fully setup `My\Atom\Feed` object, all you have to do is call the following:
+
+    $xml = $service->writeValueObject($feed);
+
+
+How it works
+------------
+
+When you pass a classname to `mapValueObject`, sabre/xml automatically creates
+an instance of that class when it comes across the element name you specified.
+
+Take for instance this portion of the XML document:
+
+     <entry xmlns="http://www.w3.org/2005/Atom">
+       <title>Atom-Powered Robots Run Amok</title>
+       <link href="http://example.org/2003/12/13/atom03"/>
+       <id>urn:uuid:1225c695-cfb8-4ebb-aaaa-80da344efa6a</id>
+       <updated>2003-12-13T18:30:02Z</updated>
+       <summary>Some text.</summary>
+     </entry>
+
+When we call the following function:
+
+    $service->mapValueObject('{http://www.w3.org/2005/Atom}entry', 'My\Atom\Entry'); 
+
+We are basically saying:
+
+* Map the element `entry`.
+* In the XML namespace `http://www.w3.org/2005/Atom`
+* To the PHP class `My\Atom\Entry`.
+
+In this example all the child elements (`title`, `link`, `id`) are also all in
+the same XML namespace. If this is the case, we will see if the class
+`My\Atom\Entry` has a `public` property with the same name, and set it's value.
+
+If entry had sub-elements in a different XML namespace, they would be discarded.
+
+One special trick is that if you define your class with a property and give it
+a default value that's an array, `sabre/xml` will immediately assume that more
+than one element may appear. In the above example, both `$authors` and `$entry`
+was defaulted to an empty array. This signals `sabre/xml` that multiple
+`<author>` and `<entry>` elements may appear as children.
+
+
+Under the hood
+--------------
+
+Ultimately this only works for simple mappings. As soon as your objects have
+multiple namespaces, or if you need to parse out attributes, ValueObjects are
+immediately too simplistic for you.
+
+In those cases you need to write custom serializers/deserializers for your
+objects. If you paid attention to the examples so far, you will have noted
+that the atom feed contained this element:
+
+       <link href="http://example.org/2003/12/13/atom03"/>
+
+The parser in fact discarded the `href` atttribute and it's value. The only
+way around that is to write a custom deserializer.
+
+The following example demonstrates how you would parse `<link>`. First, we
+need a class representing atom links:
+
+    namespace My\Atom;
+    class Link {
+
+        public $href;
+        public $rel;
+        public $type;
+        public $hrefLang;
+        public $title;
+        public $length;
+
+    }
+
+And now, a custom deserializer, defined on the Service:
+
+    $service->elementMap['{http://www.w3.org/2005/Atom}link'] = function($reader) {
+
+        $link = new My\Atom\Link();
+        foreach($reader->parseAttributes() as $key=>$value) {
```

Comment:
Should we also provide a deserializer function which puts attributes onto a object properties?